### PR TITLE
Attempt to fix coveralls

### PR DESCRIPTION
**Pull request by mzjp2**
Let's see if this works...

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] AppVeyor and Travis CI passes
* [ ] Test coverage is 100% (Coveralls passes)

### CODE REVIEW

**Comment by jakirkham** [comment]
From the Coveralls docs on Travis Pro, it says to do this..


> First, if you are using Travis CI , make sure it's building correctly. If you're using Travis Pro, add the following lines to a .coveralls.yml file in your repository's root:
>
> ```
> service_name: travis-pro
> repo_token: f9ihk8Pur6zETkTohfPmOPwbEWrZTkKK8
> ```

**Comment by jakirkham** [comment]
Are you able to see [this page]( https://coveralls.io/github/zarr-developers/zarr-python )?

**Comment by mzjp2** [comment]
> From the Coveralls docs on Travis Pro, it says to do this..
> 
> > First, if you are using Travis CI , make sure it's building correctly. If you're using Travis Pro, add the following lines to a .coveralls.yml file in your repository's root:
> > ```
> > service_name: travis-pro
> > repo_token: f9ihk8Pur6zETkTohfPmOPwbEWrZTkKK8
> > ```

That's the generic advice when not using a coveralls client like we are (hence the use of the `.coveralls.yml` file. We're using https://github.com/coveralls-clients/coveralls-python and they had this tip in their docs: https://coveralls-python.readthedocs.io/en/latest/usage/configuration.html

**Comment by jakirkham** [comment]
FWIW I regenerated the token already 😉

It seems they mention `COVERALLS_REPO_TOKEN` should be set. We might need `travis encrypt` to obscure this value.

**Comment by mzjp2** [comment]
It seems to say that it'll find the token automatically: 🤷‍♂ how it does that, I don't know

> If coveralls-python is being run on TravisCI, it will automatically set the token for communication with coveralls.io. Otherwise, you should set the environment variable COVERALLS_REPO_TOKEN, which can be found on the dashboard for your project in coveralls.io:

**Comment by mzjp2** [comment]
Aha, running `coveralls debug` in `travis` yields:

> coverage.misc.CoverageException: Couldn't use data file '/home/travis/build/zarr-developers/zarr-python/.coverage': Looks like a coverage 4.x data file. Are you mixing versions of coverage?

**Comment by jakirkham** [comment]
Good catch! Maybe we can update [this line]( https://github.com/zarr-developers/zarr-python/blob/994f2449b84be544c9dfac3e23a15be3f5478b71/requirements_dev_optional.txt#L14 )?

File: requirements_dev_optional.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ redis==3.3.8
 pymongo==3.9.0
 # optional test requirements
 tox==3.14.0
-coverage==4.5.4
+coverage==5.0.3
 coveralls==1.8.2
```
Comment (jakirkham) [inline requirements_dev_optional.txt]: Maybe we can update this too?

```suggestion
coveralls==1.11.1
```

Comment (mzjp2) [reply requirements_dev_optional.txt]: Yep, was just looking at this. Just waiting to see if this works or not, then going to try that. :)

Comment (mzjp2) [reply requirements_dev_optional.txt]: Okay, looks like it worked with `debug` (but the service name is potentially wrong, let's see...).

**Comment by mzjp2** [comment]
Looks like the Python 3.5 (the one that's finished) got submitted to coveralls.io successfully at least.

**Comment by jakirkham** [comment]
Happy with merging to `master`. Perhaps ( https://github.com/lemurheavy/coveralls-public/issues/1408 ) is just indicating a service outage.

**Comment by mzjp2** [comment]
it works! :)

image

**Comment by jakirkham** [comment]
Thanks for working on this @mzjp2! 😄

**Comment by jakirkham** [comment]
Bumping coverage a bit with PR ( https://github.com/zarr-developers/zarr-python/pull/542 ).

**Comment by jakirkham** [comment]
If you have a chance @mzjp2, would you be able to look at [the numcodecs repo]( https://github.com/zarr-developers/numcodecs ) as well? Guessing it needs similar changes.

**Comment by mzjp2** [comment]
Done @jakirkham, should be all good now, good spot. :)

**Comment by jakirkham** [comment]
Wow! That was fast. Thanks Zain! 😄

**Pull request merged**